File: wrappers/dotnet/indy-sdk-dotnet-test/Wrapper/AgentTests/AgentAddIdentityTest.cs
﻿using Indy.Sdk.Dotnet.Test.Wrapper.AgentTests;
using Indy.Sdk.Dotnet.Wrapper;
using Microsoft.VisualStudio.TestTools.UnitTesting;
using System.Threading.Tasks;

namespace Indy.Sdk.Dotnet.Test.Wrapper.AgentTests
{
    [TestClass]
    public class AgentAddIdentityTest : AgentIntegrationTestBase
    {
        [TestMethod]
        public async Task TestAgentAddIdentityWorks()
        {
            var endpoint = "127.0.0.1:9601";

            var myDidResult = await Signus.CreateAndStoreMyDidAsync(_wallet, "{}");

            var activeListener = await Agent.AgentListenAsync(endpoint, _incomingConnectionObserver);

            await activeListener.AddIdentityAsync(_pool, _wallet, myDidResult.Did);
        }

        [TestMethod]
        public async Task TestAgentAddIdentityWorksForMultiplyKeys()
        {
            var endpoint = "127.0.0.1:9602";

            var myDid1 = await Signus.CreateAndStoreMyDidAsync(_wallet, "{}");
            var myDid2 = await Signus.CreateAndStoreMyDidAsync(_wallet, "{}");

            CreateAndStoreMyDidResult[] didResults = { myDid1, myDid2 };

            var activeListener = await Agent.AgentListenAsync(endpoint, _incomingConnectionObserver);

            foreach (var didResult in didResults)
            {
                await activeListener.AddIdentityAsync(_pool, _wallet, didResult.Did);
            }
        }
    }
}

File: wrappers/java/src/test/java/org/hyperledger/indy/sdk/agent/AgentAddIdentityTest.java
package org.hyperledger.indy.sdk.agent;

import org.hyperledger.indy.sdk.agent.Agent.Listener;
import org.hyperledger.indy.sdk.signus.Signus;
import org.hyperledger.indy.sdk.signus.SignusResults;
import org.junit.Test;


public class AgentAddIdentityTest extends AgentIntegrationTest {

	@Test
	public void testAgentAddIdentityWorks() throws Exception {
		String endpoint = "127.0.0.1:9601";

		SignusResults.CreateAndStoreMyDidResult myDid = Signus.createAndStoreMyDid(wallet, "{}").get();

		Listener activeListener = Agent.agentListen(endpoint, incomingConnectionObserver).get();

		activeListener.agentAddIdentity(pool, wallet, myDid.getDid()).get();
	}

	@Test
	public void testAgentAddIdentityWorksForMultiplyKeys() throws Exception {
		String endpoint = "127.0.0.1:9602";

		SignusResults.CreateAndStoreMyDidResult myDid1 = Signus.createAndStoreMyDid(wallet, "{}").get();
		SignusResults.CreateAndStoreMyDidResult myDid2 = Signus.createAndStoreMyDid(wallet, "{}").get();

		SignusResults.CreateAndStoreMyDidResult[] dids = {myDid1, myDid2};

		Listener activeListener = Agent.agentListen(endpoint, incomingConnectionObserver).get();

		for (SignusResults.CreateAndStoreMyDidResult did : dids) {
			activeListener.agentAddIdentity(pool, wallet, did.getDid()).get();
		}
	}
}